packages/eslint-plugin/src/rules/restrict-plus-operands.ts

```diff
@@ -70,6 +70,11 @@ export default util.createRule<Options, MessageIds>({
         return types.every(value => value === types[0]) ? types[0] : 'invalid';
       }
 
+      if (type.isIntersection()) {
+        const types = type.types.map(getBaseTypeOfLiteralType);
+        return types.some(value => value === 'string') ? 'string' : 'invalid';
+      }
```

Comment:
it took me a second to figure out why this works for valid cases.
For the weird cases we care about (like `{} & string`), typescript tells us the type is an intersection type.

For other cases like `unknown & string`, typescript tells us the type is just `string`!
And for bad cases like `number & string`, typescript again tells us the type is just `never`!

This is pretty awesome! 😄